instrumentation/opentelemetry-instrumentation-threading/tests/test_threading.py

```diff
@@ -0,0 +1,203 @@
+from opentelemetry.test.test_base import TestBase
+
+
+class TestThreading(TestBase):
+    def setUp(self):
+        super().setUp()
+        self._tracer = self.tracer_provider.get_tracer(__name__)
+        self._mock_span_contexts: List[trace.SpanContext] = []
+        ThreadingInstrumentor().instrument()
+
+    def tearDown(self):
+        ThreadingInstrumentor().uninstrument()
+        super().tearDown()
+
+    def get_root_span(self):
+        return self._tracer.start_as_current_span("rootSpan")
+
+    def test_trace_context_propagation_in_thread(self):
+        self.run_threading_test(threading.Thread(target=self.fake_func))
+
+    def test_trace_context_propagation_in_timer(self):
+        self.run_threading_test(
+            threading.Timer(interval=1, function=self.fake_func)
+        )
+
+    def run_threading_test(self, thread: threading.Thread):
+        with self.get_root_span() as span:
+            expected_span_context = span.get_span_context()
+            thread.start()
+            thread.join()
+
+            # check result
+            self.assertEqual(len(self._mock_span_contexts), 1)
+            self.assert_span_context_equality(
+                self._mock_span_contexts[0], expected_span_context
+            )
+
+    def test_trace_context_propagation_in_thread_pool(self):
+        max_workers = 10
+        executor = ThreadPoolExecutor(max_workers=max_workers)
+
+        expected_span_contexts: List[trace.SpanContext] = []
+        futures_list = []
+        for num in range(max_workers):
+            with self._tracer.start_as_current_span(f"trace_{num}") as span:
+                expected_span_context = span.get_span_context()
+                expected_span_contexts.append(expected_span_context)
+                future = executor.submit(self.fake_func)
+                futures_list.append(future)
+
+        for future in as_completed(futures_list):
+            future.result()
+
+        # check result
+        self.assertEqual(len(self._mock_span_contexts), max_workers)
+        self.assertEqual(
+            len(self._mock_span_contexts), len(expected_span_contexts)
+        )
+        for index, mock_span_context in enumerate(self._mock_span_contexts):
+            self.assert_span_context_equality(
+                mock_span_context, expected_span_contexts[index]
+            )
+
+    def fake_func(self):
+        span_context = trace.get_current_span().get_span_context()
+        self._mock_span_contexts.append(span_context)
+
+    def assert_span_context_equality(
+        self,
+        result_span_context: trace.SpanContext,
+        expected_span_context: trace.SpanContext,
+    ):
+        self.assertEqual(result_span_context, expected_span_context)
+        self.assertEqual(
+            result_span_context.trace_id, expected_span_context.trace_id
+        )
+        self.assertEqual(
+            result_span_context.span_id, expected_span_context.span_id
+        )
```

Comment:
These checks are already covered by L93 and can be removed

Reply:
Ok, I think I can remove the `assert_span_context_equality` method and only compare the `span_context`, right?

Reply:
I have modified it, thanks.